content/rest/messages.textile

```diff
@@ -157,6 +157,12 @@ bc[swift]. channel.publish("event", data: "This is my payload")
 
 bc[go]. channel.Publish("event", "This is my payload")
 
+h4(#idempotent). Enabling idempotent publishing
+
+It is possible for a client publishing through REST to not receive an acknowledgement of receipt from Ably for numerous reasons, which can lead to the client re-publishing a message. If you wish to ensure that Ably can recognise these re-sent messages as duplicates, you should "enable idempotent publishing":/rest/usage#client-options. This attaches an identifier to each message sent from the client, enabling Ably to detect duplicates of the same message and ignore them.
+
+If you wish to use your own IDs for idempotent publishing, you can specify it when you "publish a message":/rest/channels#publish as the "id":#id element of the "Message":#properties object. At present sending multiple messages in a single publish with client-specified "ids":#id is "not fully supported":https://support.ably.io/solution/articles/3000088357-how-can-i-publish-multiple-messages-in-a-single-rest-request-with-idempotency.
```

Comment:
"If you wish to provide a unique identifier each message published, all publishes will be idempotent - that is you can freely publish the message from one or more workers without worrying about duplicates. When "publishing a message":/rest/channels#publish, if you specify the "id":#id element of the "Message":#properties object, the publish operation will be idempotent.  Please note that publishing multiple messages in a single publish operation with client-specified "ids":#id "has many constraints you need to be aware of":https://support.ably.io/solution/articles/3000088357-how-can-i-publish-multiple-messages-in-a-single-rest-request-with-idempotency."

Reply:
@tomczoink I think we should have closed code examples here to be honest, see https://docs.google.com/document/d/1WGvzS_O8B1AXOfQNmAnCadTeUysUGlTZbTC-nNkpzFo/edit# for how simple this is.  We could even for now just do one language if you want to keep it very simple.